Adds Interfaces and Stub Impl for Scheduler

cppwfs opened the pull request:
resolves #1

cppwfs commented on spring-cloud-scheduler-spi/src/main/java/org/springframework/cloud/scheduler/spi/core/ScheduleRequest.java:
```diff
@@ -0,0 +1,150 @@
+ * artifact, and any deployment properties.
+ *
+ * Deployment properties are related to a specific implementation of the SPI
+ * and will never be passed into an app itself. For example, a runtime container
```
1) Had a chance to look at each of the deployer implementations.  They use the deployer props for setting up the app to be deployed but do not pass the deployer props on to the app itself as environment variables.  
2) Yes I think they would be useful, but we would want to have a flag that would add that to the AppDefinition properties.

cppwfs commented:
Updated README to remove references to implementations.

markpollack commented on spring-cloud-scheduler-spi/src/main/java/org/springframework/cloud/scheduler/spi/core/ScheduleRequest.java:
```diff
@@ -0,0 +1,121 @@
+ 	 * @param schedulerProperties properties that contain scheduler specific informaton.
+	 * @param scheduleName the name associated with the schedule.
+	 */
+	public ScheduleRequest(AppDefinition definition, Map<String, String> schedulerProperties, String scheduleName, Resource resource) {
```
I think we can get rid of this constructor.  We don't have such a constructor in AppDeploymentRequest (without deployer properties) so not sure why we need it here.

markpollack commented on spring-cloud-scheduler-spi/src/main/java/org/springframework/cloud/scheduler/spi/implementation/CreateScheduleException.java:
```diff
@@ -0,0 +1,29 @@
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.scheduler.spi.implementation;
```
Exceptions are usually placed in the `root` of the package, in this case I think we should put them in `org.springframework.cloud.scheduler.spi`

markpollack commented on spring-cloud-scheduler-spi/src/main/java/org/springframework/cloud/scheduler/spi/implementation/SchedulerStub.java:
```diff
@@ -0,0 +1,51 @@
+ *
+ * @author Glenn Renfro
+ */
+public class SchedulerStub implements Scheduler {
```
Don't think we need this class.

cppwfs replied:
I disagree.  This will be used by downstream DataFlow Servers that will need to implement a Scheduler bu t have not had a chance to do it yet.  
i.e. Local and K8's.

markpollack commented:
I suggest moving all exceptions into `core` as well as `Scheduler` and `ScheduleInfo`

markpollack commented on spring-cloud-scheduler-spi/src/main/java/org/springframework/cloud/scheduler/spi/implementation/DuplicateScheduleException.java:
```diff
@@ -0,0 +1,30 @@
+ *
+ * @author Glenn Renfro
+ */
+public class DuplicateScheduleException extends RuntimeException{
```
Not clear to me which method in `Scheduler` would throw this.  `CreateScheduleException` and `UnScheduleException` pair with the methods `schedule` and `unschedule`.

markpollack commented:
If we have these two (or more) exceptions related to scheduling, there should be a base `ScheduleException` so that we can have a hierarchy.

markpollack commented on spring-cloud-scheduler-spi/src/main/java/org/springframework/cloud/scheduler/spi/Scheduler.java:
```diff
@@ -0,0 +1,67 @@
+	 * @param taskDefinitionName to retrieve {@link ScheduleInfo}s for a specified taskDefinitionName.
+	 * @return A List of {@link ScheduleInfo}s configured for the provided taskDefinitionName.
+	 */
+	List<ScheduleInfo> list(String taskDefinitionName) ;
```
We should indicate what type of exception is thrown in case of failure.  If we have a general `ScheduleException` that take a basic message, then perhaps we can use that .... or come up with another exception type?

cppwfs commented:
* Created Base Exception for Scheduler.  All scheduler exceptions inherit from the base `SchedulerException`.
* Requested Classes moved to core
* Removed unused ScheduleRequest constructor
* Removed DuplicateScheduleException

markpollack commented:
squashed and merged as fb086bd09d75f5ebaf4ac59211a2529230795f61

On second look, perhaps we move the exception classes up one level.  Will discuss another time.
